packages/frontend/src/components/markdown-editor/markdown-editor.tsx

```diff
@@ -65,19 +67,51 @@ const emojiCompleter = {
 interface Props {
   contents: string;
   onContentsChange: (updatedValue: string) => any;
+  uploadFile?: (blob: File, name: string) => any;
```

Comment:
Let's improve the function result type. It's something like `Promise<UploadSingleFileResult>`, using the GraphQL-generated type for that mutation result.  You can probably get the exact type from the mutation call site, or you can look through the generated types file.